react-frontend-app/src/App.js

```diff
@@ -32,42 +35,85 @@ function App() {
-    console.log({surname}, {zipcode});
 
-    axios.get(`http://localhost:5000/surgeo?surname=${surname}&zipcode=${zipcode}`)
-      .then(res => console.log(res.data));
+    setVariable(await axios.get(`http://localhost:5000/surgeo?surname=${surname}&zipcode=${zipcode}`)
+      .then(res => {
+        setVariable(res.data);
+        return res.data;
+      }));
+    setLoading(false);
   }
 
-  return (
-    <div className="App">
-      <MainNavBar />
-      <br/><br/><br/><br/><br/>
-      <h1>Zest AI Race Predictor Prototype</h1>
-      <br/>
-      <Description />
-      <form onSubmit={handleSubmit}>
-        <label>Please enter your information:</label>
-        <br/><br/>
-        Surname: <input type="text" onChange={handleSurname}/>
-        <br/><br/>
-        Zip code: <input type="text" onChange={handleZip}/>
-        <br/><br/>
-        <Button type="submit" value="Submit">submit</Button>
-      </form>
-      <br/>
-      <div>
-        <h3>Breakdown</h3>
-        <BarGraph />
+  if (variable == "this is the default variable" || loading) {
+    return (
+      <div className="App">
+        <MainNavBar />
+        <br /><br /><br /><br /><br />
+        <h1>Zest AI Race Predictor Prototype</h1>
+        <br />
+        <Description />
+        <form onSubmit={handleSubmit}>
+          <label>Please enter your information:</label>
+          <br /><br />
+          Surname: <input type="text" onChange={handleSurname} />
+          <br /><br />
+          Zip code: <input type="text" onChange={handleZip} />
+          <br /><br />
+          <Button type="submit" value="Submit">submit</Button>
+        </form>
+        <br />
+        <div>
+          Nothing here, submit your data!
+        </div>
+
+        {/* Also in progress */}
+        {/* <Display /> */}
+        <Footer />
       </div>
+    );
+  }
+  else {
+    return (
+      <div className="App">
+        <MainNavBar />
+        <br /><br /><br /><br /><br />
+        <h1>Zest AI Race Predictor Prototype</h1>
+        <br />
+        <Description />
+        <form onSubmit={handleSubmit}>
+          <label>Please enter your information:</label>
+          <br /><br />
+          Surname: <input type="text" onChange={handleSurname} />
+          <br /><br />
+          Zip code: <input type="text" onChange={handleZip} />
+          <br /><br />
+          <Button type="submit" value="Submit">submit</Button>
+        </form>
+        <br />
+        <div>
+          <h3>Breakdown</h3>
+          {/* <BarGraph testProps={variable} /> */}
+          <BarGraph
+            // zcta={variable.zcta[0]}
+            // name={variable.name[0]}
+            white={variable.white[0]}
+            black={variable.black[0]}
+            api={variable.api[0]}
+            hispanic={variable.hispanic[0]}
+            multiple={variable.multiple[0]}
+            native={variable.native[0]}
+          />
+          {/* <BarGraph /> */}
+        </div>
```

Comment:
IMPT: It looks like the only difference between the if/else statement return values is the "Breakdown" section (ie: lines 96-108). Everything else is duplicated, which makes this set-up design slightly redundant. This code duplication makes the code more difficult to maintain and prone to bugs. For example, if someone wanted to update the "Zest AI Race Predictor Prototype" title, they would have to remember to make the exact same changes in both the "if" return statement and the "else" return statement. To summarize, the current set-up is really vulnerable to the two versions getting out of sync.

Can you instead factor out the "Breakdown" section into a variable "content" that is conditionally set to either "Nothing here, submit your data" or "<h3>Breakdown...etc"?

Reply:
In fact, I think you could take advantage of Javascript's conditional operator to do something like this:

\<div>
    {Nothing here, submit your data! ? (variable == "this is the default variable" || loading) :
         \<h3>Breakdown... etc}
\</div>

Reply:
Essentially, in the above code snippet, the code within the {} is Javascript using the following format:

valueIfConditionIsTrue ? condition : valueIfConditionIsFalse

(end Javascript)

Here's some more documentation here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

Reply:
This is much cleaner, also another thing in my PR that wasn't great. Ternary operator seems much more clean!